queries.sql

```diff
@@ -7,4 +7,75 @@ SELECT date_of_birth from animals WHERE name IN ('Agumon', 'Pikachu');
 SELECT name, escape_attempts from animals WHERE weight_kg > 10.5;
 SELECT * from animals WHERE neutered =  true;
 SELECT * from animals WHERE name != 'Gabumon';
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+
+UPDATE animals 
+SET species = 'unspecified';
+
+UPDATE animals SET species = 'digimon'
+WHERE name like '%mon';
+
+UPDATE animals SET species = 'pokemon'
+WHERE species = 'unspecified';
+
+BEGIN;
+
+SAVEPOINT SP1;
+
+DELETE FROM animals;
+ROLLBACK TO SP1 ;
+
+COMMIT;
+
+-- OTHER TRANSACTION.
+
+
+BEGIN;
+-- Delete all animals born after Jan 1st, 2022.
+DELETE FROM animals WHERE date_of_birth > 'January 1, 2022';
+-- Create a savepoint for the transaction.
+SAVEPOINT ANIMAL_DELETED;
+-- Update all animals' weight to be their weight multiplied by -1.
+UPDATE animals SET weight_kg = weight_kg * -1;
+-- Rollback to the savepoint
+ROLLBACK TO SAVEPOINT ANIMAL_DELETED;
+-- Update all animals' weights that are negative to be their weight multiplied by -1.
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+-- Commit transaction
+COMMIT;
+
+
+-- AGGREGATIONS AND GROUP BY
+
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(*) FROM animals WHERE escape_attempts  = 0;
+SELECT AVG(weight_kg) AS AVERAGE_WEIGHT FROM animals;
+SELECT neutered, count(*) FROM animals GROUP BY neutered;
+SELECT species, MIN(weight_kg), MAX(weight_kg) FROM animals GROUP BY species;
+SELECT species, ROUND(AVG(escape_attempts), 0) AS "average number of escape attempts" FROM animals WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31' GROUP BY species;
+
+
+-- What animals belong to Melody Pond?
+SELECT animals.name FROM animals  JOIN owners ON animals.owner_id = owners.id WHERE owners.full_name = 'Melody Pond';
+
+-- List of all animals that are pokemon (their type is Pokemon).
+SELECT animals.name FROM animals JOIN species ON animals.species_id = species.id WHERE species.name = 'Pokemon';
```

Comment:
* Kindly correct these mistakes i pointed out above previously as you dont have a `name column in your species table rather you have `full_name` so you should say `species.full_name` instead of `species.name` . Please do this correction everywhere you have made the mistake 🙏

Reply:
@Davidosky007  No problem but can you tell me how you are having an output in the file rather than in the terminal? I guess you are using some EXTENSION. Let me know please it will save me time.

Reply:
Sure you can chat me up on slack when you are free so we setup a call for this